kafka-streams/src/main/java/io/micronaut/configuration/kafka/streams/KafkaStreamsFactory.java

```diff
@@ -42,11 +41,13 @@
 @Factory
 public class KafkaStreamsFactory implements Closeable {
 
-    private final Collection<KafkaStreams> streams = new ConcurrentLinkedDeque<>();
+    private final Map<KafkaStreams, ConfiguredStreamBuilder> streams = new ConcurrentHashMap<>();
```

Comment:
Does `KafkaStreams` implement `hashCode()`

Reply:
Why do you need access to the builder in this map.  Aren't all the configurations that the `KafkaStreams` was built with available there?

Reply:
Unfortunately the builder has the properties accessible that were used to build.  The configuration properties are NOT accessible from the KafkaStreams object directly as it is `private final StreamsConfig config;private final StreamsConfig config;`. The ONLY way to pluck out the application.id or client.id is to tie th streams to the configuration builder.  This is what Spring Boot does as well.  I used their pattern as a guide.